### PR TITLE
Shows project name dialog box when uploading a project with an invalid name.

### PR DESCRIPTION
**Pull request by AKandAK**
#2453 
This Pull request Shows project name dialog when uploading a project with an invalid name.
It doesn't suggest a name as of now.

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

File: appinventor/appengine/src/com/google/appinventor/client/youngandroid/TextValidators.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
   private static final int MAX_FILENAME_SIZE = 100;
   private static final int MIN_FILENAME_SIZE = 1;
+  public static String INVALID_PROJECT_NAME_TYPE;
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/youngandroid/TextValidators.java:26]: Rather than creating a public static field to hold an error, I would strongly recommend we move to using an `enum` to capture the various error conditions (with one `SUCCESS = 0` entry). Then, the error messages should be internationalized.

File: appinventor/appengine/src/com/google/appinventor/client/wizards/RequestNewProjectNameWizard.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2009-2011 Google, All Rights reserved
+// Copyright 2011-2020 MIT, All rights reserved
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/wizards/RequestNewProjectNameWizard.java:3]: No need to have the Google copyright here since this is new code. MIT is sufficient (and should start at 2021).

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Comment by AKandAK** [comment]
Thanks @ewpatton for the review. I have committed the changes suggested by you.

File: appinventor/appengine/src/com/google/appinventor/client/wizards/RequestNewProjectNameWizard.java [hunk 3]
```diff
@@ -46,7 +46,7 @@ public RequestNewProjectNameWizard(RequestProjectNewNameInterface newName,
     final DialogBox db = new DialogBox(false, true);              
     db.setText(title);                                                   // title of the dialog box
     db.setStyleName("ode-DialogBox");
-    db.setAnimationEnabled(true);
+    // db.setAnimationEnabled(true);
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/wizards/RequestNewProjectNameWizard.java:49]: 1. What is the reason for this change?
2. If you're going to disable code, it is preferred to remove it rather than comment it out.

Comment (AKandAK) [reply appinventor/appengine/src/com/google/appinventor/client/wizards/RequestNewProjectNameWizard.java:49]: 1.When the dialog box is closing the animation doesn't look smooth compared to the animation when dialog box opens.
After seeing other dialog boxes of the ai2 (like the one in create new project) , I thought it is better to disable the animation like other dialog boxes. 
2.I will remove the commented code.

File: appinventor/appengine/src/com/google/appinventor/client/wizards/ProjectUploadWizard.java [hunk 4]
```diff
@@ -58,48 +58,134 @@ public void execute() {
...
-            return;
-          }
+          if (!TextValidators.checkNewProjectName(filename, true)) {
+         
+            String suggestedName = "";
+            String title;
+            filename = filename.replace(" ", "_").replaceAll("[^a-zA-Z0-9_]", "");
+
+            if (TextValidators.isTitleReserved()) {
+              title = MESSAGES.reservedTitleFormatError() + " : " + filename;
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/wizards/ProjectUploadWizard.java:68]: I would recommend making the titles take arguments so that when translators translate them into other language the position of the filename can be moved if it makes sense to do so.

File: appinventor/appengine/src/com/google/appinventor/client/wizards/TemplateUploadWizard.java [hunk 5]
```diff
@@ -349,27 +349,70 @@ public void execute() {
...
+            } else {
+              suggestedName = ProjectUploadWizard.getSuggestedName(selectedTemplateNAME);
+              title = MESSAGES.duplicateTitleFormatError();
+              if (!TextValidators.checkNewProjectName(suggestedName, true)) {
+                  suggestedName = "";
+                  title += " : " + selectedTemplateNAME;
+              } else {
+                title += MESSAGES.suggestNameTitleCaption();
+              }
+            }
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/wizards/TemplateUploadWizard.java:381]: Given that a lot of this code seems a duplicate of ProjectUploadWizard you may want to refactor the logic into one of the other utility classes and just call that from here.

**Review by ewpatton — CHANGES_REQUESTED** [review]
A few more things to address but this is fairly close to being done.

**Comment by AKandAK** [comment]
@ewpatton Thanks for the Review, I have made the changes.

**Comment by ewpatton** [comment]
@AKandAK I think is good. Can you resolve the conflict with master so we can proceed with merging it?

**Comment by AKandAK** [comment]
@ewpatton Done

**Review by ewpatton — APPROVED** [review]
LGTM
